src/dev/build/tasks/bin/scripts/kibana

```diff
@@ -26,4 +26,8 @@ if [ -f "${CONFIG_DIR}/node.options" ]; then
   KBN_NODE_OPTS="$(grep -v ^# < ${CONFIG_DIR}/node.options | xargs)"
 fi
 
-NODE_OPTIONS="--no-warnings --max-http-header-size=65536 $KBN_NODE_OPTS $NODE_OPTIONS" NODE_ENV=production exec "${NODE}" "${DIR}/src/cli/dist" "${@}"
+# Enable capturing heap snapshots. See https://nodejs.org/api/cli.html#--heapsnapshot-signalsignal
+# For now only on POSIX platforms, term signals work differently on Windows.
+NODE_OPTIONS_HEAPSNAPSHOT_DEFAULT="--heapsnapshot-signal=SIGUSR2 --diagnostic-dir=./data"
```

Comment:
On deb and rpm package this isn't going to work.  The `kibana` user will not have write access to anything in `/usr/share/kibana`.  The `data` directory is at `/var/lib/kibana`.

Reply:
For tar.gz it should be something like this to support starting Kibana from outside the root directory:
```suggestion
NODE_OPTIONS_HEAPSNAPSHOT_DEFAULT="--heapsnapshot-signal=SIGUSR2 --diagnostic-dir=${DIR}/data"
```

Reply:
https://github.com/elastic/kibana/blob/4b4c573904b35650701a695767d560ee4f3917f3/src/dev/build/tasks/os_packages/service_templates/env/kibana:

```
NODE_OPTIONS_HEAPSNAPSHOT_DEFAULT="--heapsnapshot-signal=SIGUSR2 --diagnostic-dir=/var/lib/kibana"
```

and then in bin/kibana:

```
NODE_OPTIONS_HEAPSNAPSHOT_DEFAULT="${NODE_OPTIONS_HEAPSNAPSHOT_DEFAULT:---heapsnapshot-signal=SIGUSR2 --diagnostic-dir=${DIR}/data}" 
```

Mind if we add the `ci:build-all-platforms` and `ci:build-os-packages` labels after to test if we go this route?